packages/charm/src/commands.ts

```diff
@@ -21,10 +21,13 @@ export const castSpellAsCharm = async (
     if (!recipe) return;
 
     console.log("Casting...");
+    const tx = charmManager.runtime.edit();
```

Comment:
is there a reason we wouldn't put this edit/commit at the start/end of runPersistent?

Reply:
I ended up making `tx` optional for read-only operations, but require it for anything that can write.